netket/graph/lattice.py

```diff
@@ -491,37 +445,48 @@ def basis_coords(self) -> CoordT:
 
     # Site lookup
     # ------------------------------------------------------------------------
+
+    class InvalidSiteError(Exception):
+        pass
+
+    class InvalidWaveVectorError(Exception):
+        pass
+
```

Comment:
Can you not scope those inside of the class?

Reply:
They're never used in other contexts, so I didn't want to litter the interface of `graph` with it.

Reply:
I think it's fine to have the exceptions defined here.

Reply:
If they are here they litter the autocompletion dialogs that is something people use to discover functionality.

And I find little reason to just prepend an underscore to their name (they will still litter the namespace).

I would put them in this file, at the top and not reexport them

```
>>> g = nk.graph.Grid([3,4])
>>> g.
g.adjacency_list(             g.coordinates                 g.id_from_basis_coords(       g.n_edges                     g.position_from_basis_coords( g.sites                       g.vector_to_site(
g.atom_label(                 g.distances(                  g.id_from_position(           g.n_nodes                     g.positions                   g.space_group(
g.atoms_coord                 g.draw(                       g.InvalidSiteError(           g.ndim                        g.rotation_group(             g.space_group_builder(
g.automorphisms(              g.edges(                      g.InvalidWaveVectorError(     g.nodes(                      g.site_offsets                g.to_reciprocal_lattice(
g.basis_coords                g.extent                      g.is_bipartite(               g.pbc                         g.site_to_coord(              g.translation_group(
g.basis_vectors               g.graph                       g.is_connected(               g.point_group(                g.site_to_vector(             g.vector_to_coord(
>>> g.Invalid
```

Reply:
> If they are here they litter the autocompletion dialogs that is something people use to discover functionality.
> 
> And I find little reason to just prepend an underscore to their name (they will still litter the namespace).

The underscore would work IMO

> I would put them in this file, at the top and not reexport them

They are needed in the tests, so they have to be visible outside of `lattice.py`...

Reply:
It's a very internal thing.
And in tests you can just do 
`from netket.graph._lattice import InvalidSiteError, InvalidWaveVectorError `